modules/service/topologyspreadconstraint/topology_spread_constraint.k

```diff
@@ -0,0 +1,107 @@
+        As a result, when the number of eligible domains is less than minDomains, scheduler won't schedule more than maxSkew Pods to those
+        domains. If value is nil, the constraint behaves as if MinDomains is equal to 1. Valid values are integers greater than 0. When value
+        is not nil, WhenUnsatisfiable must be DoNotSchedule.
+        For example, in a 3-zone cluster, MaxSkew is set to 2, MinDomains is set to 5 and pods with the same labelSelector spread as
+        2/2/2: | zone1 | zone2 | zone3 | | P P | P P | P P | The number of domains is less than 5(MinDomains), so "global minimum" is treated
+        as 0. In this situation, new pod with the same labelSelector cannot be scheduled, because computed skew will be 3(3 - 0) if new Pod is
+        scheduled to any of the three zones, it will violate MaxSkew.
+        This is a beta field and requires the MinDomainsInPodTopologySpread feature gate to be enabled (enabled by default).
+    nodeAffinityPolicy: str, default is Undefined, optional.
+        NodeAffinityPolicy indicates how we will treat Pod's nodeAffinity/nodeSelector when calculating pod topology spread skew. Options are:
+        - Honor: only nodes matching nodeAffinity/nodeSelector are included in the calculations. - Ignore: nodeAffinity/nodeSelector are ignored.
+        All nodes are included in the calculations.
+        If this value is nil, the behavior is equivalent to the Honor policy. This is a beta-level feature default enabled by the
+        NodeInclusionPolicyInPodTopologySpread feature flag.
+    nodeTaintsPolicy: str, default is Undefined, optional.
+        NodeTaintsPolicy indicates how we will treat node taints when calculating pod topology spread skew. Options are: - Honor: nodes without
+        taints, along with tainted nodes for which the incoming pod has a toleration, are included. - Ignore: node taints are ignored. All nodes
+        are included.
+        If this value is nil, the behavior is equivalent to the Ignore policy. This is a beta-level feature default enabled by the
+        NodeInclusionPolicyInPodTopologySpread feature flag.
+
+    Examples
+    --------
+    import catalog.workload.topologyspreadconstraint as tp
+
+    topologyspreadconstraint = tp.TopologySpreadConstraint {
+        maxSkew: 1
+        topologyKey: "kubernetes.io/hostname"
+        whenUnsatisfiable: "DoNotSchedule"
+    }
+    """
+
+    # MaxSkew describes the degree to which pods may be unevenly distributed.
+    # Must be greater than 0.
+    maxSkew:                      int
+
+    # TopologyKey is the key of node labels.
+    topologyKey:                  str
+
+    # WhenUnsatisfiable indicates how to deal with a pod if it doesn't satisfy the spread constraint.
+    whenUnsatisfiable:            str
+
+    # LabelSelector is used to find matching pods.
+    labelSelector?:               LabelSelector
```

Comment:
Is `labelSelector` optional? It seems `matchLabelKeys` relies on this field described in K8s doc.
To determine which pods should be considered for topology spreading, a `labelSelector` or both `labelSelector` and `matchLabelKeys` must be specified.

Reply:
I  think labelSelector and  matchLabelKeys are no mandatory binding relationship. K8s define labelSelector and matchLabelKeys are optional, so i follow k8s the definition.

![image](https://github.com/user-attachments/assets/672c73e8-b2b1-456d-a3f7-6b5e7e554526)

Reply:
Add matchLabelKeys check if labelSelector exist.